### PR TITLE
add new package : rocketmq

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/rocketmq/package.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    version('4.5.1', sha256='0c46e4b652b007d07e9c456eb2e275126b9210c27cd56bee518809f33c8ed437')
+    version('4.5.0', sha256='d75dc26291b47413f7c565bc65499501e3499f01beb713246586f72844e31042')
+    version('4.4.0', sha256='8a948e240e8d2ebbf4c40c180105d088a937f82a594cd1f2ae527b20349f1d34')
+    version('4.3.2', sha256='e31210a86266ee218eb6ff4f8ca6e211439895459c3bdad162067b573d9e3415')
+
+    depends_on('java@8:', type='run')
+
+    def url_for_version(self, version):
+        url = "https://archive.apache.org/dist/rocketmq/{0}/rocketmq-all-{0}-bin-release.zip"
+        version = version.string.split('-')[0]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rocketmq/package.py]: This line isn't needed, right?

Comment (darmac) [reply var/spack/repos/builtin/packages/rocketmq/package.py]: We tried remove it, and found that the version will be '4.6.0-bin',but we expected it as '4.6.0'.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/rocketmq/package.py]: See #14490

Comment (darmac) [reply var/spack/repos/builtin/packages/rocketmq/package.py]: Got it, thanks.

**Pull request merged**